src/Inventory/Request.php

```diff
@@ -349,18 +364,126 @@ public function inventory($data)
+    */
+    public function handleNetDiscoveryTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_NETDISCOVERY_TASK, $params);
+
+        return $params['options']['response'][self::NETDISCOVERY_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled netinventory task support on contact request
+    *
+    * @return array
+    */
+    public function handleNetInventoryTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_NETINVENTORY_TASK, $params);
+
+        return $params['options']['response'][self::NETINV_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled ESX task support on contact request
+    *
+    * @return array
+    */
+    public function handleESXTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_ESX_TASK, $params);
+
+        return $params['options']['response'][self::ESX_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled collect task support on contact request
+    *
+    * @return array
+    */
+    public function handleCollectTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_COLLECT_TASK, $params);
+
+        return $params['options']['response'][self::COLLECT_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled deploy task support on contact request
+    *
+    * @return array
+    */
+    public function handleDeployTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_DEPLOY_TASK, $params);
+
+        return $params['options']['response'][self::DEPLOY_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled wakeonlan task support on contact request
     *
     * @return array
     */
-    public function handleInventoryTask()
+    public function handleWakeOnLanTask(): array
     {
 
         $params['options']['response'] = [];
         $params['item'] = $this->inventory->getAgent();
-        $params = Plugin::doHookFunction("handle_inventory_task", $params);
+        $params = Plugin::doHookFunction(Hooks::HANDLE_WAKEONLAN_TASK, $params);
 
-        return $params['options']['response'];
+        return $params['options']['response'][self::WAKEONLAN_TASK] ?? [];
+    }
+
+   /**
+    * Handle agent enabled remoteinventory task support on contact request
+    *
+    * @return array
+    */
+    public function handleRemoteInventoryTask(): array
+    {
+       // Permit remoteinventory task as it could have been setup on agent side
+        return [
```

Comment:
Indeed, here, I should probably modify the agent as `handleRemoteInventoryTask()` should be better there to say the server can tell the agent it knows some remoteinventory tasks has been setup on server-side. On the agent side, if a remote inventory is setup, it should simply be considered as an inventory task from the server point of view, not a remoteinventory one.
So you're right and we need to avoid that "strange" answer. I'll update GLPI-Agent accordingly and rework this part.

Reply:
Resolved in 441904c. GLPI-Agent required an update as explained.